src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -489,46 +492,52 @@ Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std
     }
 
     auto config = orig_config;
-
-    CNNNetwork clonedNetwork = InferenceEngine::details::cloneNetwork(network);
+    const std::shared_ptr<ov::Model> cloned_model = model->clone();
     const bool enableLPT = shouldEnableLPT(config, engConfig);
-    auto nGraphFunc = clonedNetwork.getFunction();
-    Config::ModelType modelType = getModelType(nGraphFunc);
+    Config::ModelType modelType = getModelType(model);
     ov::element::Type inferencePrecision = getInferencePrecision(config, engConfig, modelType);
-    const Config::SnippetsMode snippetsMode = getSnippetsMode(config, engConfig); 
-
-    DEBUG_LOG(PrintableModel(*nGraphFunc, "org_"));
+    const Config::SnippetsMode snippetsMode = getSnippetsMode(config, engConfig);
+    DEBUG_LOG(PrintableModel(*cloned_model, "org_"));
 
-    Transformations transformations(nGraphFunc, enableLPT, inferencePrecision, isLegacyAPI(), snippetsMode, engConfig);
+    Transformations transformations(cloned_model, enableLPT, inferencePrecision, is_legacy_api(), snippetsMode, engConfig);
     transformations.UpToLpt();
 
     if (!is_cpu_map_available()) {
-        ApplyPerformanceHints(config, nGraphFunc);
+        apply_performance_hints(config, cloned_model);
     }
 
     // update the props after the perf mode translated to configs
     // TODO: Clarify the behavior of SetConfig method. Skip eng_config or not?
     Config conf = engConfig;
-
     conf.readProperties(config, modelType);
-    CalculateStreams(conf, nGraphFunc);
+    calculate_streams(conf, cloned_model);
 
     transformations.PostLpt();
     transformations.Snippets();
 
     // need to check that all outputs have static shapes
     // checking that all inputs have static shapes is performed in the common part
-    if (isLegacyAPI()) {
-        for (const auto& res : nGraphFunc->get_results()) {
+    if (is_legacy_api()) {
+        for (const auto& res : cloned_model->get_results()) {
             if (res->get_input_partial_shape(0).is_dynamic()) {
-                IE_THROW() << "CPU plug-in can't load a model with dynamic output shapes via legacy API.";
+                OPENVINO_THROW("CPU plug-in can't load a model with dynamic output shapes via legacy API.");
             }
         }
     }
 
     transformations.CpuSpecificOpSet();
+    DEBUG_LOG(PrintableModel(*cloned_model, "cpu_"));
 
-    DEBUG_LOG(PrintableModel(*nGraphFunc, "cpu_"));
+    if ((cloned_model->inputs().size() != model->inputs().size()) ||
+        (cloned_model->outputs().size() != model->outputs().size())) {
+        OPENVINO_THROW("Input/output port size mismatched!");
```

Comment:
Could you please write more clear error message to help better understand the error nature by an external observer,